src/state/session/index.tsx

```diff
@@ -276,20 +275,24 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
         createPersistSessionHandler(
           account,
           ({expired, refreshedAccount}) => {
-            upsertAccount(refreshedAccount, expired)
+            if (expired) {
+              clearCurrentAccount()
+            } else {
+              setCurrentAccount(refreshedAccount)
+            }
```

Comment:
All persist handlers should handle token expiration AND update e.g. refresh token is used to get a new access token